projects/map-ol/src/lib/map-ol.component.ts

```diff
@@ -158,20 +161,35 @@ export class MapOlComponent implements OnInit, AfterViewInit, AfterViewChecked,
     } else {
       /** if layers already on the map -length not changed- update them */
       for (const layer of layers) {
-        const ollayer = this.mapSvc.getLayerByKey({ key: 'id', value: layer.id }, type);
+        const ollayer = this.mapSvc.getLayerByKey({ key: 'id', value: layer.id }, type) as olBaseLayer | olLayer<any> | olLayerGroup;
         if (ollayer) {
           if (ollayer.getVisible() !== layer.visible) {
             ollayer.setVisible(layer.visible);
           }
           if (ollayer.getOpacity() !== layer.opacity) {
             ollayer.setOpacity(layer.opacity);
           }
-          if (layer instanceof CustomLayer && ollayer.getSource()) {
+          if (layer instanceof CustomLayer && ollayer instanceof olLayer) {
             const newSource = layer.custom_layer.getSource();
             const oldSource = ollayer.getSource();
             if (newSource && getUid(oldSource) !== getUid(newSource)) {
               ollayer.setSource(newSource);
             }
+          } else if (layer instanceof CustomLayer && layer.custom_layer instanceof olLayerGroup && ollayer instanceof olLayerGroup) {
+            const newLayers = layer.custom_layer.getLayers().getArray();
+            const oldLayers = ollayer.getLayers().getArray();
+
+            /** assume the order and length of layers is not changing and no more grouping!!! */
```

Comment:
Just a note for the future: this line might actually profit from `utils-maps/ol/flattenLayers` :)

Reply:
The problem here is that we may have to detect changes in the array of a LayerGroup and to do this we have to deep compare every object because in a CustomLayer we don't know what is in there.
Normally when we create layers we set the Layer.id on the openlayers object as well, but on a custom group we dont create the layers so we cannot compare the ids which maybe are not added.

Flatten the layers would only be helpful if someone uses a CustomLayer with an openlayers layer group which as well uses a layer group.

Also i think this is a really rare problem because this functionality (where we update the layer source) is mostly useful for vector layers where the source is changing after they where added to the map e.g the grid in the demo maps event example.